docs/source/cc_launcher.md

```diff
@@ -0,0 +1,301 @@
+
+Configuring the peer to use external builders involves adding a configuration
+block to `core.yaml` that defines external builders. Each external builder
+definition must include a name (used for logging) and the path to parent of
+the `bin` directory containing the builder scripts.
+
+An optional list of environment variable names to propagate from the peer when
+invoking the external builder scripts can also be provided.
+
+The following example defines two external builders:
+
+```yaml
+chaincode:
+  externalBuilders:
+  - name: my-golang-builder
+    path: /builders/golang
+    environmentWhitelist:
+    - GOPROXY
+    - GONOPROXY
+    - GOSUMDB
+    - GONOSUMDB
+  - name: noop-builder
+    path: /builders/binary
+```
+
+In this example, the implementation of "my-golang-builder" is contained within
+the `/builders/golang` directory and its build scripts are located in
+`/builders/golang/bin`. When the peer invokes any of the build scripts
+associated with "my-golang-builder", it will not propagate the values of any
+environment variables in the whitelist.
+
+> Note: The following environment variables are always propagated to external
+> builders:
+> - LD_LIBRARY_PATH
+> - LIBPATH
+> - PATH
+> - TMPDIR
+
+When an `externalBuilder` configuration is present, the peer will iterate over
+the list of builders in the order provided, invoking `bin/detect` until one
+completes successfully. If no builder completes `detect` successfully, the
+peer will fallback to using the legacy Docker build process implemented within
+the peer. This means that external builders are completely optional.
+
+In the example above, the peer will attempt to use "my-golang-builder",
+followed by "noop-builder", and finally the peer internal build process.
+
+# Chaincode Packages
+
+As part of the new lifecycle introduced with Fabric 2.0, the chaincode package
+format changed from serialized protocol buffer messages to a gzip compressed
+POSIX tape archive. Chaincode packages created with `peer lifecycle chaincode
+package` use this new format.
+
+## Lifecycle Chaincode Package Contents
+
+A lifecycle chaincode package contains two files. The first file,
+`code.tar.gz` is a gzip compressed POSIX tape archive. This file includes the
+source artifacts for chaincode. Packages created by the peer CLI will place
+the chaincode implementation source under the `src` directory and chaincode
+metadata (like CouchDB indexes) under the `META-INF` directory.
+
+The second file, `metadata.json` is a JSON document with three keys:
+- `type`: the chaincode type (e.g. GOLANG, JAVA, NODE)
+- `path`: for go chaincode, the GOPATH or GOMOD relative path to the main
+  chaincode package; undefined for other types
+- `label`: the chaincode label
+
+## Chaincode Packages and External Builders
+
+When a chaincode package is installed to a peer, the contents of `code.tar.gz`
+and `metadata.json` are not processed prior to calling external builders. This
```

Comment:
Except that the label is used by the new lifecycle for computing package id

Reply:
Done.